app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt

```diff
@@ -69,7 +75,23 @@ class EventsFragment : Fragment() {
             it?.let { showProgressBar(it) }
         })
 
-        eventsViewModel.loadEvents()
+        /*if (eventsViewModel.showAllEvents) {
+            eventsViewModel.loadEvents()
+            eventsViewModel.showAllEvents = false
+        }*/
```

Comment:
@iamareebjamal All events are loaded if I uncomment this when the app starts(as expected). But when I put the location of the event and search then if the event is already present in the list of events, all the events in the table are displayed  otherwise if I am searching the event for the first time correct events are displayed.

So what I am doing right now is just searching the events according to location when the user enters the location similar to how it is happening in the Search fragment. I am a little confused about  what should be the correct approach here

Reply:
Yes always

Reply:
So that means they may also always launch online query and never show events from database?

Reply:
Yes exactly ! I will update the PR with the required changes